### PR TITLE
test: reframe BAT test_prompts goal-first — state goals, not tool names

### LINKED ISSUE #311: [feature] BAT agent-facing prompts should state goals, not name tools
BAT tests are run by an orchestrating/grading agent: it reads the entry, sets it up, feeds the `test_prompt` to a **sub-agent under test**, then grades whether that sub-agent found and used the right tool. The whole point is *discovery* — so the `test_prompt` (the part handed to the sub-agent) must state the scenario + goal in plain language and **not name any MCP tool, gateway, or arg**.

The rest of the entry — title, `## Expected`, `**Failure modes**`, criteria — **does** name the tool(s): the grading agent needs to know which tool the sub-agent was supposed to pick in order to score the run. Tool names belong in the setup/grading scaffolding, never in the prompt fed to the sub-agent.

Today ~60 of 316 `test_prompt`s name the tool/gateway + args (e.g. *"Call the hub_export_custom_rule tool to export rule X"*) — that tests transcription, not discovery.

**Scope:** reframe those `test_prompt`s goal-first while keeping tool names in the criteria. Edge case (per-test): tests whose subject *is* tool-routing/error-handling may still need to name the surface in the prompt. `T703` (shipped with `hub_call_device_replace`) is an example written the right way.

### PR DESCRIPTION
**Pull request by kingpanther13**
﻿## Summary

- Reframes every BAT `test_prompt` that named an MCP tool, gateway, or transcribed call args into a plain-language scenario + goal, so the suite tests tool *discovery* rather than transcription — 154 prompts total (83 in `tests/BAT-v2.md`, 71 in `tests/BAT-rm-native-crud.md`). Tool names stay in each test's title/**Expected** grading text.
- 23 prompts intentionally still name the surface because their subject IS the surface (wrong-gateway routing, tool-name-off-tools/list routing, mutually-exclusive or deliberately malformed call shapes, gateway-override catalog visibility) — the per-test edge case the issue carves out.
- Codifies the rule as a "Prompt style — goal-first" section in both active BAT files, and fixes the now-stale "prompts use the shipped tool names" note in `BAT-rm-native-crud.md`. `tests/BAT.md` (superseded v1) was audited and left untouched as a historical snapshot — it's essentially compliant (1 real violation, 11 carve-outs).

## Type of change

- [ ] `feat` — new feature or capability
- [ ] `fix` — bug fix
- [ ] `chore` — maintenance, dependency bump, or housekeeping
- [ ] `refactor` — code restructure with no behaviour change
- [ ] `docs` — documentation only
- [x] `test` — tests only
- [ ] `ci` — CI/CD pipeline change

## Changes

- Closes #311.
- `tests/BAT-v2.md`: 83 `test_prompt`s reframed goal-first; new "Prompt style — goal-first" guidance under Test Format; coverage-summary wording updated ("Sections 1-9 use explicit tool references" no longer true); where a reframe displaced grading detail, it was moved into **Expected**.
- `tests/BAT-rm-native-crud.md`: 71 `test_prompt`s reframed; header tool-names note corrected; same guidance section added; **Expected** blocks updated to carry the tool paths the prompts no longer name.
- Setup/teardown prompts untouched (orchestrator scaffolding — allowed to name tools). No test IDs or titles changed (`e2e_test.py` references T642/T703 by ID).

## Release Notes

- Acceptance-test suite improvements only — no functional changes to the app.

## Testing

- `python tests/sandbox_lint.py` — all checks passed (BAT-v2 is in scope of its tool-count doc check).
- Markdown-only change to manual test-scenario docs; no runtime surface. Verified the two carve-out prompts remaining in `BAT-rm-native-crud.md` and ~21 in `BAT-v2.md` are all deliberate surface-subject tests.

## Checklist

- [ ] **Unit tests added for any new MCP tools, regressions, or bug fixes** (N/A — docs-only change to test scenarios)
- [ ] **e2e tests added for new tools and/or regression tests added for any bug fix** (N/A — no tool behaviour change)
- [x] Sandbox lint passes: `python tests/sandbox_lint.py`
- [ ] `./gradlew test` passes locally (or CI confirms — no Groovy touched; CI will confirm)
- [x] Live-hub BAT tests updated if tool behaviour changed (this PR is the BAT update)
- [x] Documentation updated if user-facing behaviour or tool surface changed (guidance sections added)
- [ ] New/renamed MCP tools follow `AGENTS.md` Tool Design Rules (N/A)

### CODE REVIEW

File: tests/BAT-rm-native-crud.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Supplement to `tests/BAT-v2.md`. Scenarios in this file exercise the native Rule
 - `hub_manage_native_rules_and_apps.hub_set_rule_private_boolean`
 - `hub_read_apps_code.hub_get_app_config` (used as fallback verification)
 
-**Note on tool names:** prompts use the shipped tool names — `hub_set_rule` for RM create/edit (omit appId to create, pass appId to edit), `hub_delete_native_app` for delete, and `hub_get_app_config` (in `hub_read_apps_code`) for readback.
+**Note on tool names:** the expected tool paths are `hub_set_rule` for RM create/edit (omit appId to create, pass appId to edit), `hub_delete_native_app` for delete, and `hub_get_app_config` (in `hub_read_apps_code`) for readback. These names live in each test's title and **Expected** grading text — the `test_prompt`s themselves stay goal-first (see Prompt style below).
```
[the remark this answers is not in the document]
Comment (kingpanther13) [reply tests/BAT-rm-native-crud.md:19]: Done — added a Release Notes bullet to the description. Tests-only PR, so the note just flags there's no functional change.
